### DISCUSSION TITLE
Gas requirement of function YOUR_FUNCTION is infinite: If the gas requirement of a function is higher than the block gas limit, it cannot be executed. Please avoid loops in your functions or actions that modify large areas of storage (this includes clearing or copying arrays in storage)

### DISCUSSION THREAD

**Discussion by Samk13** [smartcontractkit/full-blockchain-solidity-course-py#813] [Q&A]
In Remix IDE if you get this warning and you want to get rid of it:
![image](https://user-images.githubusercontent.com/36583694/149836243-34492268-88ed-4a53-afd6-772a8c915978.png)

The solution that clears the warning:

change `string` types to `bytes32` so you limit the string size or any bytes value.

but when you deploy you get a revert error when testing the functionalities (SimpleStorage as an example)
How can you fix the warning?

**Comment by cromewar** [comment]
Nice Tip  @Samk13 this is very helpful.
